packages/transports/http/src/index.ts

```diff
@@ -23,11 +35,68 @@ export const getSubgraphExecutor: TransportExecutorFactoryFn<'http', HTTPTranspo
     if (Array.isArray(transportEntry.headers)) {
       headers = Object.fromEntries(transportEntry.headers);
     }
-    return buildHTTPExecutor({
+
+    const httpExecutor = buildHTTPExecutor({
       endpoint: transportEntry.location,
       headers,
       fetch,
       print: printFnForHTTPExecutor,
       ...transportEntry.options,
     });
+    const wsExecutors: { [hash: string]: Executor } = {};
+
+    return function HTTPExecutor(request) {
+      if (request.operationType === 'subscription') {
+        const wsOpts =
```

Comment:
Why don't we create these executors outside this function?

Reply:
Because the executors map keys depend on the auth request header. While hashing the executor, it uses url+token.

Reply:
Ok ok I saw it after. My bad 👍